### PR TITLE
Clang CI option

### LINKED ISSUE #201: Cleanup CI
As discussed, prior to the 3.0 release we will clean up the CI:

- Drop U18 as a target
- Drop Python2 support
- Add a build against a more modern gcc
- Add a build against clang

### PR DESCRIPTION
**Pull request by graeme-a-stewart**
For the c8-dev platform activate also a build with clang

We add an additional martix variable (compiler) that propages through to the environment, that triggers settng the C++ compiler to clang++. It would be then easy to add this compiler (or any other) for as many platforms as we think we need.

Closes #201

### CODE REVIEW

**Comment by graeme-a-stewart** [comment]
HI @amete - this is ready to go now. We have a clang build on CentOS8 and it's easy to add other platforms if we felt the need. In parallel the containers were brought up to date and they all work find in the text matrix.

File: .github/scripts/build-test.sh [hunk 1]
```diff
@@ -1,18 +1,35 @@
...
 # to cmake for particular platforms
 cd /tmp
+echo "Starting build and test for platform $PLATFORM, compiler suite $COMPILER"
 if [ -z "$CXX" ]; then
-    CXX=$(type -p g++)
+	if [ "$COMPILER" == "clang" ]; then
+		CXX=$(type -p clang++)
+	else
+		# gcc suite is the fall through
+    	CXX=$(type -p g++)
```
Comment (amete) [inline .github/scripts/build-test.sh:16]: Admittedly this is my OCD kicking in but can we indent this properly? 😄

**Review by amete — CHANGES_REQUESTED** [review]
LGTM. I have only a minor cosmetic comment 👍

File: .github/scripts/build-test.sh (hunk 1 above)
Comment (graeme-a-stewart) [reply .github/scripts/build-test.sh]: It seemed to be ok for me, but it was tabs - changed to space indentation (4 spaces), so then that should be more consistent

**Review by amete — APPROVED** [review]

**Pull request merged**